refactor(scheduler): Move priority types to priorities package

sonasingh46 opened the pull request:
**What this PR does / why we need it**:
This moves the priority types from the algorithm package
to priorities package.
Idea is to move the type to the packages where it is
implemented. This will ease the future refactor process.

**What type of PR is this?**
/kind cleanup
/sig scheduling

**Which issue(s) this PR fixes**:
This fixes #71865 

**Does this PR introduce a user-facing change?**:
 "NONE"

sonasingh46 commented:
/assign @misterikkit

wgliang commented on pkg/scheduler/algorithm/priorities/types.go:
```diff
@@ -0,0 +1,58 @@
+/*Copyright 2016 The Kubernetes Authors.
```
2016-> 2018

sonasingh46 replied:
Done

wgliang commented:
/ok-to-test
/release-note-none

denkensk commented on pkg/scheduler/algorithm/priorities/types_test.go:
```diff
@@ -14,14 +14,14 @@ See the License for the specific language governing permissions and
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/pkg/scheduler/algorithm"
 	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+	"testing"
```
move "testing" to top

misterikkit reviewed (changes requested):
You will need to update all the call sites to use the new import path, and then run `./hack/update-bazel.sh` to fix the BUILD files.

misterikkit reviewed (changes requested):
lgtm overall

one minor comment about import alias

please squash commits and then let's get it approved and merged.

(People might not be around next week to approve, but who knows?)

sonasingh46 commented on pkg/scheduler/core/extender_test.go:
```diff
@@ -25,6 +25,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/kubernetes/pkg/scheduler/algorithm"
+	algorithmpriorities "k8s.io/kubernetes/pkg/scheduler/algorithm/priorities"
```
fixed

sonasingh46 commented:
/test pull-kubernetes-integration

sonasingh46 commented:
/retest

misterikkit reviewed (approved):
/lgtm

Let's get this rebased and approved! :confetti_ball:

sonasingh46 commented:
/retest

sonasingh46 committed:
refactor(scheduler): Move priority types to priorities package

This moves the priority types from the algorithm package
to priorities package.
Idea is to move the type to the packages where it is
implemented. This will ease the future refactor process.

sonasingh46 commented:
/retest

wgliang commented:
/lgtm

misterikkit reviewed (approved):
/lgtm
/assign @bsalamat @k82cn for approval

ravisantoshgudimetla reviewed (approved):
Thanks @sonasingh46 for the cleanup

/approve
